minetest/minetest
# Implement `minetest.get_node` and `minetest.get_node_or_nil` with LuaJIT FFI

Since these methods do no call back to Lua or do memory allocation (i.e. throw exceptions), they can be implemented with FFI. This gives a significant speed up and increases LuaJIT optimization potential.

## To do

This PR is Ready for Review.

## How to test

- Run the unit tests.
- Run a bunch of mods to see that `minetest.get_node` and `minetest.get_node_or_nil` still work.
- Run the benchmark `/bench_get_node`. The time I got without FFI was roughly 300ms, while the time I got with FFI enabled was roughly 210ms. This benchmark is purposefully not the best-case scenario for the FFI version.
